### DISCUSSION TITLE
Comparing SHAP values across models and re-trained models

### DISCUSSION THREAD

**Discussion by alanzablocki** [shap/shap#2965] [General]
**Can we compare SHAP values (feature importance and attribution) across different tree based models?** 

The idea would be to have different models trained with same data to understand what the model is learning both overall and for each training instance? I feel like this is being done in this post https://medium.com/towards-data-science/interpretable-machine-learning-with-xgboost-9ec80d148d27 to show inconsistency? To what extent can you compare feature attributions across models?

**Can we compare or subtract SHAP values between an old and a model re-trained on the most recent data (trained model last month, re-trained this month)?**
In this example, we have the same model type, same features, but now the new data picks up on something that happened in the real world and so the feature attribution might change. The person examining the prediction attribution would be able to say that for the prediction this month feature X1 is more important overall vs feature X2 ( which was most important last month) because something drastic happened in the real world that affects feature X1. Would taking the differences between the mean SHAP value be the correct thing to do, to show the extent of how much that feature now adds to the prediction? I feel like this is bordering on causality, so I am curious if this is something that SHAP can answer, or if SHAP should not be used to answer.

Thank you.

**Comment by dsgibbons** [comment]
This feels more like a discussion rather than an issue.
